src/AutoMapper/Execution/TypeMapPlanBuilder.cs

```diff
@@ -93,8 +93,9 @@ private void CheckForCycles(HashSet<TypeMap> typeMapsPath)
             typeMapsPath.Add(_typeMap);
             var defaultPropertyMap = new PropertyMap(default(MemberInfo), null);
             var properties = 
-                _typeMap.GetPropertyMaps().Where(pm=>pm.CanResolveValue()).Select(pm=>new { PropertyTypeMap = ResolvePropertyTypeMap(pm), PropertyMap = pm })
-                .Concat(_typeMap.ConstructorMappingTypes.Select(tp => new { PropertyTypeMap = ResolveTypeMap(tp), PropertyMap = defaultPropertyMap }))
+                _typeMap.GetPropertyMaps().Concat(_typeMap.PathMaps.Select(pm => new PropertyMap(pm))).Where(pm=>pm.CanResolveValue())
```

Comment:
This is the second place we're wrapping `PathMap` with `PropertyMap`, I wonder if there's maybe a common interface or base class between the two? Then those could be exposed up at `TypeMap`?

Reply:
Yes, but I don't think that would really help. Because when building the execution plan (the other place) we really want the whole PropertyMap interface (and PathMap implements just a few things). Here probably that interface could be useful, but it's just one place at the moment.

Reply:
Probably null would work here as well :)

Reply:
No, I need the type map.

Reply:
Do you think there's anything to do with the two places in that same class where we're creating PropertyMaps from PathMaps? I guess it seems weird to me we're doing it twice.

Reply:
Do we want to rewrite CreatePropertyMapFunc so it works both with the common interface and the full property map interface? Or have a more complete interface of which PathMap implements a few methods? I think that's weird too. Doable, but...

Reply:
Meh. Just leave it for now.